presto-main/src/main/java/com/facebook/presto/SessionRepresentation.java

```diff
@@ -229,4 +231,51 @@ public Session toSession(SessionPropertyManager sessionPropertyManager)
                 sessionPropertyManager,
                 preparedStatements);
     }
+
+    private static class InternalPrincipal
+            implements Principal
+    {
+        private final String name;
+
+        private InternalPrincipal(String name)
+        {
+            this.name = requireNonNull(name, "name is null");
+        }
+
+        public static Optional<Principal> createPrincipal(Optional<String> principal)
+        {
+            return principal.map(name -> new InternalPrincipal(name));
+        }
+
+        @Override
+        public String getName()
+        {
+            return name;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            InternalPrincipal that = (InternalPrincipal) o;
+            return Objects.equals(name, that.name);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(name);
```

Comment:
Since you have only one argument you can do `Objects.hashCode(name)` which is more efficient.